kbase-extension/static/kbase/js/widgets/narrative_core/upload/stagingAreaViewer.js

```diff
@@ -253,6 +261,42 @@ define([
             this.$elem.append(errorElem);
         },
 
+
+        identifyImporterMappings: function (stagingFiles) {
+            /*
+                Add a list of top matches for each file, sorted by weight
+                 */
+            const fileNames = stagingFiles.map(f => f['path']);
+            const fileList = $.param({ 'file_list': fileNames }, true);
+            var mappings = [];
+            return Promise.resolve(
+                this.stagingServiceClient.importer_mappings(
+                    {
+                        file_list: fileList
+                    }
+                )
+            ).then(function (data) {
+                //Extract mappings, sort by weight, assign mappings to staging files
+                mappings = JSON.parse(data)['mappings'];
+                mappings.forEach(function (mapping) {
+                    try {
+                        if (mapping) {
+                            mapping.sort((a, b) => (a.app_weight < b.app_weight));
+                        }
+                    } catch (err) {
+                        console.error('Mapping is malformed', err);
```

Comment:
Does this need to do anything else, like fail and throw an Error? It seems like it just won't sort that mapping, but if it's malformed here, will that cause problems later on?

Reply:
I will remove the try/catch and make it fail then